CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword
+
+```csharp
+// Bad
+String.IsNullOrEmpty(name);
+
+// Good
+string.IsNullOrEmpty(name);
+```
+
+### Use readonly fields when possible
+
+### Use string interpolation
+
+```csharp
+// Bad
+var message = string.Format("My name is {0} {1}.", person.FirstName, person.LastName);
+
+// Good
+var message = $"My name is {person.FirstName} {person.LastName}.");
+```
+
+## Layout
+
+### Use *four spaces* per indentation level
+
+```csharp
+// Bad
+public void SomeMethod()
+{
+  DoSomethingFirst();
+  DoSomerthingLater();
+}
+
+// Good
+public void SomeMethod()
+{
+    DoSomethingFirst();

```

Comment:
Agree with @AlphaGit.

Maybe something like this code can work...


```csharp
// Bad
public void SomeMethod()
{
 ・DoSomethingFirst();
 ・DoSomethingLater();
}

// Good
public void SomeMethod()
{
 ・・・DoSomethingFirst();
 ・・・DoSomerthingLater();
}
```


I would also add a link to the IDE documentation on how to configure the identation. Many people forgot about this.

I also use the IDE with the "View White Space" option enabled. The keyboard shortcut is CTRL+R, CTRL+W.